src/api/app/models/project.rb

```diff
@@ -129,6 +129,8 @@ def deleted?(project_name)
     end
 
     def restore(project_name, backend_opts = {})
+      return nil if find_by_name(project_name)
```

Comment:
@saraycp maybe its better to use the `get_by_name` method as stated in the comment above the `find_by_name` method https://github.com/openSUSE/open-build-service/blob/2d22e0c11b71c4bfbcd97701f9535c9c77039123/src/api/app/models/project.rb#L228 you could then rescue from the exception it throws when there is no project

Reply:
I want to "undelete" a project that was previously deleted. If I use `get_by_name`, an exception is raised because it doesn't find the object (obviously), so it won't perform the rest of the code of "undelete".

Anyway, I've realized the solution is not going to be straightforward. I'm going to close this PR and simply introduce this change: https://github.com/openSUSE/open-build-service/pull/11932